web-ui/src/pages/PermissionsPage.jsx

```diff
@@ -66,14 +66,28 @@ const EditPermissionsPage = () => {
   const { csrf } = state;
   const roles = selectRoles(state);
   const hasPermission = selectHasPermissionAssignmentPermission(state);
-  const [selectedRole, setSelectedRole] = useState(
-    roles && roles.find(() => true)
-  );
+  const [selectedRole, setSelectedRole] = useState(roles[0]);
   const [categoriesList, setCategoriesList] = useState([]);
   const [rolePermissionsList, setRolePermissionsList] = useState([]);
   const [rolePermissions, setRolePermissions] = useState([]);
   const [refresh, setRefresh] = useState(true);
 
+  useEffect(() => {
+    const url = new URL(location.href);
+    const roleName = url.searchParams.get('role');
+    const role = roles.find(r => r.role === roleName);
+    setSelectedRole(role || roles[0]);
+  }, []);
+
+  useEffect(() => {
```

Comment:
Every time the user changes the selected role, this updates the query parameter in the URL.